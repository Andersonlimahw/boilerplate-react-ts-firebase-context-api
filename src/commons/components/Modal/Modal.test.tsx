import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Modal } from './';

describe('Modal', () => {
    test('renders with default props', () => {
        const { getByText } = render(<Modal id="test" title="Test Modal" content={<p>Test content</p>} isOpen onClose={() => {}} />);
        const title = getByText(/Test Modal/i);
        const content = getByText(/Test content/i);
        const closeButton = getByText(/close/i);

        expect(title).toBeTruthy();
        expect(content).toBeTruthy();
        expect(closeButton).toBeTruthy();
    });

    test('renders with onSubmit prop', () => {
        const handleSubmit = jest.fn();
        const { getByTestId } = render(<Modal id="test" title="Test Modal" content={<p>Test content</p>} isOpen onClose={() => {}} onSubmit={handleSubmit} />);
        const submitButton = getByTestId('modal-submit-button');

        expect(submitButton).toBeTruthy();
        fireEvent.click(submitButton);
        expect(handleSubmit).toHaveBeenCalledTimes(1);
    });

    test('matches snapshot', () => {
        const { asFragment } = render(<Modal id="test" title="Test Modal" content={<p>Test content</p>} isOpen onClose={() => {}} />);
        expect(asFragment()).toMatchSnapshot();
    });
});