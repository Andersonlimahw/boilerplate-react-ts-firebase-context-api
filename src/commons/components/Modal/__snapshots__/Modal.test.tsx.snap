// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
<DocumentFragment>
  <dialog
    class="modal"
    id="test"
    open=""
  >
    <div
      class="modal_container"
    >
      <div
        class="modal_title"
      >
        <h2>
          Test Modal
        </h2>
      </div>
      <div
        class="modal_content"
      >
        <p>
          Test content
        </p>
      </div>
      <div
        class="modal_footer"
      >
        <button
          class="button"
          data-testid="modal-cancel-button"
          type="submit"
        >
          close
        </button>
      </div>
    </div>
  </dialog>
</DocumentFragment>
`;
